catalog_impls/src/table_based.rs

```diff
@@ -77,7 +78,13 @@ define_result!(Error);
 /// Table based catalog manager
 #[derive(Clone)]
 pub struct TableBasedManager {
-    inner: Arc<Inner>,
+    /// Sys catalog table
+    catalog_table: Arc<SysCatalogTable>,
+    catalogs: CatalogMap,
```

Comment:
Can this bare `HashMap` be cloned? Unlike other arc fields, cloning this may lead to inconsistent status.